nexus/reconfigurator/execution/src/omicron_zones.rs

```diff
@@ -21,8 +22,17 @@ use uuid::Uuid;
 pub(crate) async fn deploy_zones(
     opctx: &OpContext,
     sleds_by_id: &BTreeMap<Uuid, Sled>,
-    zones: &BTreeMap<Uuid, OmicronZonesConfig>,
+    zones: &BTreeMap<Uuid, BlueprintZonesConfig>,
 ) -> Result<(), Vec<anyhow::Error>> {
+    // Make this code fail to compile if a new zone state is added.
```

Comment:
I don't think this needs to change on this PR, but related to the question about whether `all_omicron_zones` should take a filter: is the point of this check that when we add a new state, the implementation may need to change in unknown ways? Is there a way to restructure the inputs to this so that we don't feel the need for this guard?

Reply:
Yeah, good question -- let me look at this a bit.

Reply:
I've changed `to_omicron_zones_config` to accept a filter argument, which feels quite satisfying.